### PR TITLE
DT-1982 Handle active/inactive building updates

### PR DESCRIPTION
**Pull request by mikehalmamoj**
There is 1 issue with these changes that doesn't quite work - hence the draft PR.

If we have multiple buildings and some have changed, the current sync processing deletes the old prison address and creates a new one (it's not possible to match prison addresses and court buildings in that case).  Fair enough.

However, this means that if we update an inactive building, as we do not know which Nomis address has been updated we lose any information about its end date - and we end up resetting the end date to today.

As far as I'm concerned it's an edge case that doesn't cause major issues - the Nomis address is still inactive and we lose an uninteresting piece of data, the true date the address was closed.  So I can live with it but would appreciate other opinions.

UPDATE: Everyone agrees the edge case is not worth worrying about.  It's not even possible to do that in the registers UI as active flag is updated independently of the main address details.

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsregisterstonomisupdate/services/CourtRegisterUpdateServiceTest.kt [hunk 1]
```diff
@@ -23,7 +27,14 @@ class CourtRegisterUpdateServiceTest {
 
   @BeforeEach
   fun before() {
-    service = CourtRegisterUpdateService(courtRegisterService, prisonService, prisonReferenceDataService, telemetryClient, true, GsonConfig().gson())
+    service = CourtRegisterUpdateService(
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsregisterstonomisupdate/services/CourtRegisterUpdateServiceTest.kt:30]: Ktlint got rather excited in this PR.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsregisterstonomisupdate/services/CourtRegisterUpdateService.kt [hunk 2]
```diff
@@ -359,7 +360,8 @@ class CourtRegisterUpdateService(
           primary = building == buildings[0], // first one in the list?
           noFixedAddress = false,
           startDate = LocalDate.now(),
-          endDate = null,
+          endDate = if (building.active.not()) { LocalDate.now() } else { null }, // TODO updating an inactive court building will update the endDate to today - we don't have the old prison address and so cannot tell if it was already inactive
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsregisterstonomisupdate/services/CourtRegisterUpdateService.kt:363]: I don't understand this bit, so probably needs a discussion

**Review by Mjwillis — APPROVED** [review]
Lgtm

**Review by andymarke — APPROVED** [review]
LGTM
